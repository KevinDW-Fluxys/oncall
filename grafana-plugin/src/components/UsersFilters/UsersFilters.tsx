import React, { ChangeEvent, useCallback } from 'react';

import { Icon, Input } from '@grafana/ui';
import cn from 'classnames/bind';

import styles from './UsersFilters.module.css';

const cx = cn.bind(styles);

interface UsersFiltersProps {
  searchTerm: string;
  onChange: (searchTerm: string) => void;
  className?: string;
}

const UsersFilters = (props: UsersFiltersProps) => {
  const { searchTerm = '', onChange, className } = props;

  const onSearchTermChangeCallback = useCallback(
    (e: ChangeEvent<HTMLInputElement>) => {
      onChange(e.currentTarget.value);
    },
    [onChange, searchTerm]
  );

  return (
    <div className={cx('root', className)}>
      <Input
        prefix={<Icon name="search" />}
        className={cx('search', 'control')}
        placeholder="Search users..."
        value={searchTerm}
        onChange={onSearchTermChangeCallback}
      />
    </div>
  );
};

export default UsersFilters;
