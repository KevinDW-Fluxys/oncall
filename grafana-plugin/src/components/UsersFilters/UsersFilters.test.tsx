import React from 'react';

import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import UsersFilters from './UsersFilters';

describe('UsersFilters', () => {
  test('it renders properly', () => {
    const component = render(<UsersFilters searchTerm="asdfasf" onChange={() => {}} />);
    expect(component.container).toMatchSnapshot();
  });

  test('It properly calls onChange', async () => {
    const searchTerm = 'hello';
    const typedTerm = ' world';
    const className = 'potato';
    const mockOnChange = jest.fn();

    render(<UsersFilters searchTerm={searchTerm} className={className} onChange={mockOnChange} />);

    await userEvent.type(screen.getByRole('textbox'), typedTerm);

    await waitFor(() => {
      expect(mockOnChange).toHaveBeenCalledWith(searchTerm + typedTerm);
    });
  });
});
