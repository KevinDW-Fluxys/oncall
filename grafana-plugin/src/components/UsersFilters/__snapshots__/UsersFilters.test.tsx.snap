// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersFilters it renders properly 1`] = `
<div>
  <div
    class="root"
  >
    <div
      class="css-xcstkt-input-wrapper search control"
      data-testid="input-wrapper"
    >
      <div
        class="css-1w5c5dq-input-inputWrapper"
      >
        <div
          class="css-mrzmip-input-prefix"
        >
          <div
            class="css-wf08df-Icon"
          >
            <svg
              class="css-eyx4do"
              height="16"
              viewBox="0 0 24 24"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M21.71,20.29,18,16.61A9,9,0,1,0,16.61,18l3.68,3.68a1,1,0,0,0,1.42,0A1,1,0,0,0,21.71,20.29ZM11,18a7,7,0,1,1,7-7A7,7,0,0,1,11,18Z"
              />
            </svg>
          </div>
        </div>
        <input
          class="css-1mlczho-input-input"
          placeholder="Search users..."
          style="padding-left: 12px;"
          value="asdfasf"
        />
      </div>
    </div>
  </div>
</div>
`;
